backend/Testing/Browser/Base/PageTest.cs

```diff
@@ -26,6 +31,21 @@ public PageTest()
     public ILocatorAssertions Expect(ILocator locator) => Assertions.Expect(locator);
     public IPageAssertions Expect(IPage page) => Assertions.Expect(page);
     public IAPIResponseAssertions Expect(IAPIResponse response) => Assertions.Expect(response);
+    /// <summary>
+    /// Consumes a defferred exception that was "thrown" in a sub-thread, and returns it
+    /// or throws if no exception of the given type is found.
+    /// </summary>
+    public Exception ExpectDeferredException<TException>() where TException : Exception
+    {
+        var exception = DeferredExceptions.FirstOrDefault(e => e is TException);
```

Comment:
should we use single? would it be an error if there were 2 exceptions when we really only expected one?

Reply:
I don't think I'll just Single though, because I'd like a more expect-like error message.